web3/auto/infura/goerli.py

```diff
@@ -0,0 +1,17 @@
+from web3 import Web3
+from web3.middleware import (
+    geth_poa_middleware,
+)
+from web3.providers.auto import (
+    load_provider_from_uri,
+)
+
+from .endpoints import (
+    INFURA_GOERLI_DOMAIN,
+    build_infura_url,
+)
+
+_infura_url = build_infura_url(INFURA_GOERLI_DOMAIN)
+
+w3 = Web3(load_provider_from_uri(_infura_url))
```

Comment:
Then you can pass in the headers here: `w3 = Web3(load_provider_from_uri(_infura_url, _headers))`